src/twobinomial.jl

```diff
@@ -0,0 +1,541 @@
+# twobinomial.jl
+# Two sample hypothesis tests for proportions
+#
+# Copyright (C) 2020 Vladimir Arnautov
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export TwoSampleBinomialTest
+
+struct TwoSampleBinomialTask
+    x1::Int
+    n1::Int
+    x2::Int
+    n2::Int
+    δ::Float64
+    ptype::Symbol
+    htype::Symbol
+    alpha::Float64
+    method::Symbol
+end
+
+struct TwoSampleBinomialTest <: HypothesisTest
+    test::TwoSampleBinomialTask
+    est
+    se
+    ci
+    result
+end
+
+function cihalpha(htype, alpha)
+    if  htype == :equivalence || htype == :superiority
+        cialpha = alpha * 2
+    elseif htype == :equality
+        cialpha = alpha
+    else
+        throw(ArgumentError("unknown hypothesis=$(htype)"))
+    end
+    cialpha
+end
+
+"""
+    TwoSampleBinomialTest(x1::Real, n1::Real, x2::Real, n2::Real, δ::Real; ptype::Symbol, htype::Symbol, alpha::Real = 0.05, method::Symbol = :default)
+
+Perform a two sample binomial hypothesis check via computing confidence intrval
+with coverage `1 - alpha` for `equality` test or `1 - 2 * alpha` for `equivalence`
+or `superiority` test.
+
+Hypothesis type `htype` are:
+- `:equality`
+- `:equivalence`
+- `:superiority`
+
+Parameters type `ptype` are:
+    - `:difference` : risk difference;
+    - `:riskratio` : risk ratio;
+    - `:oddratio` : odd ratio;
+
+For each parameters can be used  following methods.
+
+Possible values for `method` for `difference` are:
+
+  - `:mn`
+  - `:wald`
+  - `:waldcc`
+  - `:nhs`
+  - `:nhscc`
+  - `:ac`
+  - `:ha`
+  - `:mnmee`
+  - `:mover`
+
+Possible values for `method` for `riskratio` are:
+
+  - `:mn`
+  - `:cli`
+  - `:li`
+  - `:mover`
+
+Possible values for `method` for `oddratio` are:
+
+  - `:mn`
+  - `:woolf`
+  - `:awoolf`
+  - `:mover`
+
```

Comment:
Can you add bibliographic references about these methods? Given their number, a manual covering several of them would probably be enough if you don't have the original paper for each. Though I saw below you already have gathered quite a few of them. :-)